### PR TITLE
document Architecture 8 new section names

### PR DESCRIPTION
**Pull request by frivieremicroej**

### CODE REVIEW

**Comment by frivieremicroej** [comment]
@fpinel-microej It could be interesting here to use the new tab, can you please update the branch ?

**Comment by jerome-leroux** [comment]
See https://docs.microej.com/en/feature-m0081meja-186_document_new_section_names/VEEPortingGuide/coreEngine.html#link

File: VEEPortingGuide/coreEngine.rst [hunk 1]
```diff
@@ -541,8 +541,48 @@ Link
 
 Several sections are defined by the Core Engine. Each section
 must be linked by the third-party linker.
```
Comment (jerome-leroux) [inline VEEPortingGuide/coreEngine.rst:543]: Are we listing the sections defined by the Core Engine or referenced by the Core Engine?

Should we mention the `.text.*` code sections?

File: VEEPortingGuide/coreEngine.rst [hunk 2]
```diff
@@ -541,8 +541,48 @@ Link
...
+   | ``.bss.microej.features``     | System Applications         | RW          | 4          |
+   |                               | statics                     |             |            |
+   +-------------------------------+-----------------------------+-------------+------------+
+   | ``.bss.microej.heap``         | Application heap            | RW          | 4          |
+   +-------------------------------+-----------------------------+-------------+------------+
+   | ``.bss.microej.immortals``    | Application immortal heap   | RW          | 4          |
+   |                               |                             |             |            |
+   +-------------------------------+-----------------------------+-------------+------------+
+   | ``.bss.microej.runtime``      | Core Engine                 | Internal RW | 8          |
+   |                               | internal heap               |             |            |
```
Comment (jerome-leroux) [inline VEEPortingGuide/coreEngine.rst:561]: Why "Internal"? Is this to mention that it is better to locate this section in internal RAM?

Should we defined an ordered list of the sections to specify which one should be placed in priority in faster memories?

File: VEEPortingGuide/coreEngine.rst [hunk 3]
```diff
@@ -551,12 +591,12 @@ must be linked by the third-party linker.
    | ``.bss.features.installed`` | System Applications         | RW          | 4          |
    |                             | statics                     |             |            |
```
Comment (jerome-leroux) [inline VEEPortingGuide/coreEngine.rst:592]:
```suggestion
   |                             | static fields               |             |            |
```

**Review by jerome-leroux — CHANGES_REQUESTED** [review]

**Comment by fpinel-microej** [comment]
@frivieremicroej Is this what you had in minds for the tabs ? let me know

File: VEEPortingGuide/coreEngine.rst (hunk 1 above)
Comment (frivieremicroej) [reply VEEPortingGuide/coreEngine.rst:543]: Both. 
Yes, `.text` these are special executable sections that must be mentionned

File: VEEPortingGuide/coreEngine.rst [hunk 4]
```diff
@@ -541,8 +541,48 @@ Link
...
+   | ``.rodata.microej.soar``      | Application and library     | RO          | 16         |
+   |                               | code                        |             |            |
+   +-------------------------------+-----------------------------+-------------+------------+
+   | ``.text.__icetea__*``         | Core Engine                 | RX          | ISA        |
+   |                               | generated code              |             | Specific   |
+   +-------------------------------+-----------------------------+-------------+------------+
 
-.. table:: Linker Sections
+.. note::
+	Sections ``.bss.microej.runtime``, ``.bss.microej.heap`` and ``.bss.microej.immortals`` are zero-initialized at Core Engine startup.
```
[the remark this answers is not in the document]
Comment (frivieremicroej) [reply VEEPortingGuide/coreEngine.rst]: done

File: VEEPortingGuide/coreEngine.rst [hunk 5]
```diff
@@ -541,8 +541,48 @@ Link
...
+Starting from Architecture ``8.x``, sections have been renamed to follow the standard ELF naming convention.
+
+.. table:: Linker Sections (Architecture ``8.x``)
+
+   +-------------------------------+-----------------------------+-------------+------------+
+   | Section name                  | Aim                         | Location    | Alignment  |
+   |                               |                             |             | (in bytes) |
+   +===============================+=============================+=============+============+
+   | ``.bss.microej.features``     | System Applications         | RW          | 4          |
+   |                               | statics                     |             |            |
```
[the remark this answers is not in the document]
Comment (frivieremicroej) [reply VEEPortingGuide/coreEngine.rst]: done

File: VEEPortingGuide/coreEngine.rst (hunk 2 above)
Comment (frivieremicroej) [reply VEEPortingGuide/coreEngine.rst]: Good catch, I don't remember. 
TBD

File: VEEPortingGuide/coreEngine.rst [hunk 6]
```diff
@@ -541,46 +541,91 @@ Link
...
+            | ``.rodata.microej.soar``       | Application and library     | RO          | 16         |
+            |                                | code                        |             |            |
+            +--------------------------------+-----------------------------+-------------+------------+
+            | ``.text.__icetea__*``          | Core Engine                 | RX          | ISA        |
+            |                                | generated code              |             | Specific   |
+            +--------------------------------+-----------------------------+-------------+------------+
+
+        .. note::
+            
+            During its startup, the Core Engine automatically zero-initializes the sections ``.bss.microej.runtime``, ``.bss.microej.heap`` and ``.bss.microej.immortals``.
```
Comment (jerome-leroux) [inline VEEPortingGuide/coreEngine.rst:587]:
```suggestion
            During its startup, the Core Engine automatically zero-initializes the sections ``.bss.microej.runtime``, ``.bss.microej.heap``, and ``.bss.microej.immortals``. 
```

**Review by jerome-leroux — CHANGES_REQUESTED** [review]

File: VEEPortingGuide/coreEngine.rst (hunk 2 above)
Comment (jerome-leroux) [reply VEEPortingGuide/coreEngine.rst]: Link to the note does not jump at the right position. Again, it is hidden by the banner.
I ask Felix to check that.

File: VEEPortingGuide/coreEngine.rst [hunk 7]
```diff
@@ -541,46 +541,94 @@ Link
...
+            | ``.bss.features.installed`` | System Applications         | RW          | 4          |
+            |                             | static fields               |             |            |
+            +-----------------------------+-----------------------------+-------------+------------+
+            | ``.rodata.soar.features``   | System Applications code    | RO          | 4          |
+            |                             | and resources               |             |            |
+            +-----------------------------+-----------------------------+-------------+------------+
+
+        .. note::
+            
+            During its startup, the Core Engine automatically zero-initializes the sections ``ICETEA_HEAP``, ``_java_heap`` and ``_java_immortals``.
```
Comment (jerome-leroux) [inline VEEPortingGuide/coreEngine.rst:628]:
```suggestion
            During its startup, the Core Engine automatically zero-initializes the sections ``ICETEA_HEAP``, ``_java_heap``, and ``_java_immortals``. 
```

**Review by jerome-leroux — CHANGES_REQUESTED** [review]

**Review by jerome-leroux — APPROVED** [review]

**Pull request merged**